server/src/main/java/org/elasticsearch/cluster/routing/RoutingNode.java

```diff
@@ -41,6 +37,10 @@
 
     private final LinkedHashMap<ShardId, ShardRouting> shards; // LinkedHashMap to preserve order
 
+    private final LinkedHashMap<ShardId, ShardRouting> initializingShards;
```

Comment:
With the previous two comments, there is no need for this to be a map keyed by `ShardId` since we do not do any lookups. I think this is fine:

```suggestion
    private final LinkedHashSet<ShardRouting> initializingShards;
```